hermeto/core/models/input.py

```diff
@@ -120,6 +122,60 @@ def _validate_ssl_options(self) -> Self:
         return self
 
 
+class BinaryFilter(pydantic.BaseModel, extra="forbid"):
+    """Represents a filter for binary packages.
+
+    An empty filters set means "accept all" (no filtering).
+    A populated filters set means "accept only these specific items".
+    """
+
+    filters: set[str] = set()
+
+    @property
+    def is_all(self) -> bool:
+        """Return True if this filter accepts all (no filtering)."""
+        return not self.filters
+
+    def __str__(self) -> str:
+        return ":all:" if self.is_all else ",".join(sorted(self.filters))
+
+    def __repr__(self) -> str:
+        return (
+            f"<{type(self).__name__} "
+            f"filters={':all:' if self.is_all else ','.join(sorted(self.filters))}>"
```

Comment:
The [recommended way](https://docs.python.org/3/reference/datamodel.html#object.__repr__) of defining a \_\_repr\_\_() is to have it return something that
```
If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value
```
I.e. here you would want to have something like `BinaryFilter(filters=set())`. Maybe this is a sign to move `:all:` processing into the initializer?